vllm/core/block_manager_v1.py

```diff
@@ -310,13 +310,14 @@ def can_allocate(self, seq_group: SequenceGroup) -> AllocStatus:
             return AllocStatus.LATER
 
     def _allocate_sequence(self, \
-                           seq: Sequence, \
+                           seq: Optional[Sequence], \
                            ref_count: int, \
                            is_encoder_decoder: bool = True) -> BlockTable:
         # Allocate new physical token blocks that will store the prompt tokens.
-        num_prompt_blocks = seq.n_blocks
+        num_prompt_blocks = self._get_seq_num_required_blocks(seq)
 
         block_table: BlockTable = BlockTable()
+        assert seq is not None
```

Comment:
I don't think this is required? Since `num_prompt_blocks = 0` if `seq is None` anyway

Reply:
The issue is seq is optional and directly referenced on lines 328:
```
                block = self.gpu_allocator.allocate(
                    seq.hash_of_block(logical_idx),
                    seq.num_hashed_tokens_of_block(logical_idx))
```
It's not technically possible to get there if seq is none but mypy is still complaining because it doesn't care about the check in _get_seq_num_required_blocks(seq). Do you have a more straightforward suggestion?

Reply:
I see, thanks for the explanation.